src/Core/ConstraintValidator/IsUrlRewriteValidator.php

```diff
@@ -38,16 +39,16 @@
 class IsUrlRewriteValidator extends ConstraintValidator
 {
     /**
-     * @var bool
+     * @var ConfigurationInterface
      */
-    private $isAscendedCharsAllowed;
+    private $configuration;
 
     /**
-     * @param bool $isAscendedCharsAllowed
+     * @param ConfigurationInterface $configuration
      */
-    public function __construct($isAscendedCharsAllowed)
+    public function __construct(ConfigurationInterface $configuration)
```

Comment:
I'm afraid this is a BC break 😞 
https://github.com/PrestaShop/PrestaShop/blob/1.7.6.x/src/Core/ConstraintValidator/IsUrlRewriteValidator.php#L48

What we can do is
- make the 2 params optional but check that either `$isAscendedCharsAllowed` or `$configuration` is provided
- add a `@todo` to do it in next major
- create a new file but I feel like it would overdoing

Reply:
never heard that someone would use validator class directly.

You call `\Symfony\Component\Validator\Validator\ValidatorInterface` like this:

1 sample:
```php
$this->validator->validate("my url rewrite string", new \PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints\IsUrlRewrite())
```

even surfing in symfony docs I can't find samples where it is used like this:

2 sample:
```php
$validator = new \PrestaShop\PrestaShop\Core\ConstraintValidator\IsUrlRewriteValidator($myConfigurationVariable) 
```

although maybe some people use it in wrong way - Im not even sure if it would work like that. Its complex to set up all dependencies in order to start validating like in second sample ;D 

We should even cover the code even if it is being used incorrectly?

Reply:
What if I do this:
```
class MyCustomIsUrlRewriteValidator extends IsUrlRewriteValidator
```
so I have created my own URL validator because I have a shop with complex URL rules, and when I upgrade it's broken because my service `myshop.app.validator.my_custom_url_validator` cant be instantiated by the container anymore, as it the parent now requires a new constructor parameter ?

This is espescially important as we are going to push services override as the new mechanism to replace legacy overrides.

Reply:
yes you are totally correct here - indeed some people might have added this behavior already. 

But its very important to keep this test working so instead of adding TODO I'll choose the first option which is to introduce both bool type and ConfigurationInterface via constructor.